### ISSUE TITLE
[nu-validator coverage] Deferred: CSS property/value validation inside <style> and style=""

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3946]
# [nu-validator coverage] Deferred: CSS property/value validation inside `<style>` and `style=""`

## Repro

```html
<style>
body { colr: red; }
</style>
```

nu-validator: error — "CSS: 'colr': Property 'colr' doesn't exist."
markuplint (all rules enabled by the bench): 0 violations.

Fixture: `tests/external/validator/tests/html/elements/style/css-property-error-novalid.html` (`nu-only`).

## Spec scope

CSS syntax and property registry are governed by CSS specifications (CSS Syntax Level 3, CSS Values and Units, and the individual property specifications hosted by W3C CSSWG), not by HTML LS.

markuplint's tracked spec scope is HTML Living Standard + WAI-ARIA + URL Living Standard (see `.claude/skills/bench-triage/SKILL.md`: "markuplint's reference scope is HTML Living Standard + WAI-ARIA + URL Living Standard. Anything nu enforces from a WICG draft, a vendor extension, or any other spec outside that set is treated as deferred coverage — eligible for `excluded-ids.json` only if an Issue tracks the future implementation.").

CSS is therefore out of the current tracked scope. This Issue exists to track the deferred coverage: nu-validator's `nu-only` fixtures that hit CSS grammar/property validation are recorded as `deferred-CSS` in `tests/external/bench/excluded-ids.json` and flip `nu-only` → `nu-over` without landing a rule.

## Similar precedent

Issue #3942 tracks `deferred-CSP` (Content Security Policy) with the same pattern: CSP is a W3C spec outside HTML LS scope, `packages/@markuplint/html-spec/src/spec.meta.jsonc` documents an explicit fall-through of `[http-equiv=content-security-policy]` `content` to `Any`, and the CSP fixtures are recorded in `excluded-ids.json` with the `deferred-CSP` marker.

## Proposal

Two future directions (either / both):

1. **CSS grammar validation** — hook a CSS parser into a new rule that walks `<style>` element bodies and `[style]` attribute values, reporting syntax errors (mismatched braces, unterminated strings, etc.). css-tree (already a dependency of `packages/@markuplint/types` for `MediaQueryList` / `SourceSizeList` / `<general-enclosed>` handling; see the package's `package.json`) is a plausible candidate.

2. **Property/value validation** — go one step further and check property names against a registry (e.g. `mdn-data` — MDN's CC0-1.0 CSS/HTML data package, npm `mdn-data`) and property values against their grammars. This is much larger scope than syntax-only validation.

## Priority

Deferred. HTML validation is markuplint's primary purpose; CSS-inside-HTML is secondary. Filed to unblock the nu-validator coverage triage backlog (`tests/external/snapshots/diff/nu-only.json`). Additional CSS-error fixtures that surface on future bench refreshes should reuse the same `deferred-CSS` marker (per-`id` or `patterns[]` per skill guidance) with this Issue number in the `reason` field.

## Implementation notes

- This Issue is registered as the `primary` mapping for the fixture in `tests/external/bench/issue-xref.config.ts`; xref refresh (`yarn bench:xref`) is the source of truth for the current fixture set backing this Issue.

## Benchmark cross-reference

_Sync'd from `tests/external/snapshots/` (submodule=1429313, nu-validator@sha256:210bbc353b23, markuplint=5.0.0-rc.4, generated=2026-08-12 06:31:53 UTC)._

Filter: `^html\/elements\/style\/css-property-error-novalid` → **1** fixtures. Verdict tally: match-error=0, match-clean=0, ml-only=0, nu-only=0, nu-over=1.

CSS syntax and property registry are governed by CSS specifications (CSS Syntax Level 3, CSS Values and Units, individual property specs), not by HTML LS. markuplint's tracked spec scope is HTML LS + WAI-ARIA + URL LS per `.claude/skills/bench-triage/SKILL.md`, so CSS is `deferred-CSS`. One fixture is recorded per-id in `excluded-ids.json#entries[]` and flips `nu-only` → `nu-over`; it will flip to `match-error` if/when a CSS grammar-validation rule lands (css-tree, already a dependency of `packages/@markuplint/types`, is a plausible candidate for syntax-only validation). Parallel deferred spec: #3942 (deferred-CSP).

| fixture | verdict | nu id / ml rule |
| --- | --- | --- |
| `html/elements/style/css-property-error-novalid.html` | nu-over | `nv-c5efac320bbb` |

Reproduce: see [`bench-triage` skill](https://github.com/markuplint/markuplint/blob/dev/.claude/skills/bench-triage/SKILL.md) for the audit workflow.
